mindmeld/rasaconverter.py

```diff
@@ -0,0 +1,539 @@
+
+    @staticmethod
+    def __attach_handle_to_function(handle, action, file_lines):
+        for i, line in enumerate(file_lines):
+            if (len(re.findall("def {}".format(action), line)) > 0):
+                insert_line = i
+                while (file_lines[i - 1].strip() != ''):
+                    if (file_lines[i - 1] == handle):
+                        return
+                    i = i - 1
+                file_lines.insert(insert_line, handle)
+
+    @staticmethod
+    def __attach_actions_to_function(function_name, actions, file_lines):
+        current_line = 0
+        for i, line in enumerate(file_lines):
+            if (len(re.findall("def {action}", line) > 0)):
+                current_line = i
+                break
+        while (file_lines[current_line] != ""):
+            current_line += 1
+        assert file_lines[current_line] == ""
+        file_lines[current_line:current_line] = actions
+
+    def create_init(self, mindmeld_project_directory):
+        f = self.__write_init_header()
+        actions = self.__read_actions()
+        templates = self.__read_templates()
+        # Write all functions for each action
+        RasaConverter.__write_functions(actions, templates, f)
+        f.close()
+        # Read contents of current file
+        file_lines = self.__read_file_lines()
+        stories_dictionary = self.__get_stories()
+        # Loop through all stories and create intent-action relationship
+        for steps in stories_dictionary.items():
+            # Loop through steps for each story
+            for step in steps:
+                # Get intent, any entities, and actions
+                intent = step['intent']
+                entities = step['entities']
+                actions = step['actions']
+                # attach handle to correct function
+                app_handle_string = RasaConverter.__get_app_handle(intent, entities)
+                RasaConverter.__attach_handle_to_function(app_handle_string, actions[0], file_lines)
+                # check if more than 1 action per intent
+                if len(actions) > 1:
+                    RasaConverter.__attach_actions_to_function(actions[0], actions[1:], file_lines)
+        # write all lines back to file
+        with open(mindmeld_project_directory + "/__init__.py", "w") as f:
+            f.writelines(file_lines)
+
+    @staticmethod
+    def create_main(mindmeld_project_directory):
+        string = '''# -*- coding: utf-8 -*-
+"""This module defines the command line interface for this app. Use
+`python -m <app_name>` to see available commands.
+"""
+
+if __name__ == '__main__':
+    from . import app
+    app.cli()'''
+        with open(mindmeld_project_directory + "/__main__.py", "w") as f:
+            f.write(string)
+            f.close()
+
+    @staticmethod
+    def create_config(mindmeld_project_directory):
+        string = '''# -*- coding: utf-8 -*-
+"""This module contains a template MindMeld app configuration"""
```

Comment:
we can store the string content in file and read from it

Reply:
Should I just create the python file and then copy it over to the mindmeld directory? Or should I store it as a string?

Reply:
Just keep it as text file